fury/ui.py

```diff
@@ -3707,41 +3649,75 @@ def select(self, item, multiselect=False, range_select=False):
             multi_select is True.
 
         """
-        selection_idx = self.values.index(item.element)
+        actual_value = self.get_actual_value(item.element)
+        selection_idx = self._values.index(actual_value)
         if self.multiselection and range_select:
             self.clear_selection()
             step = 1 if selection_idx >= self.last_selection_idx else -1
             for i in range(self.last_selection_idx,
                            selection_idx + step,
                            step):
-                self.selected.append(self.values[i])
+                self.selected.append(self._values[i])
 
         elif self.multiselection and multiselect:
-            if item.element in self.selected:
-                self.selected.remove(item.element)
+            if actual_value in self.selected:
+                self.selected.remove(actual_value)
             else:
-                self.selected.append(item.element)
+                self.selected.append(actual_value)
             self.last_selection_idx = selection_idx
 
         else:
             self.clear_selection()
-            self.selected.append(item.element)
+            self.selected.append(actual_value)
             self.last_selection_idx = selection_idx
 
         self.on_change()  # Call hook.
         self.update()
 
+    def compress_values(self, values):
+        """ Compressing the listboxitem element to fit into textbox
+
+        Parameters
+        ----------
+        values: :class:`ListBox2D`
+        """
+        compressed_names = []
+        textblock_width = self.slots[0].textblock.size[0]
+        for value in values:
+            char_width = textblock_width - self.margin
+            permissible_chars = int(self.slot_width)//char_width
+            total_chars = len(str(value))
+            if total_chars > permissible_chars:
+                excess_chars = total_chars - permissible_chars
+                wrapped_value = value[:int(-excess_chars) - 3] + "..."
+                compressed_names.append(wrapped_value)
+            else:
+                compressed_names.append(value)
+        return compressed_names
+
+    def set_values(self, values):
+        self._values = values
+        self.compressed_values = self.compress_values(values)
+
+    def get_actual_value(self, compressed_value):
```

Comment:
same question here

Reply:
When someone clicks on ListBoxItem then the value showing at UI is assigned to item.element.
We need the actual value that's why I created this function.Suppose we have an item named as 
"Guardians Of The Galaxy (720p) Blue Ray" and because of text overflow we could see only 
" Guardians Of The Gal...".Now here when we select the element we will get the later string.
So to get the actual value I made this function.